app/src/main/java/de/westnordost/streetcomplete/quests/surface/SidewalkSurfaceAnswer.kt

```diff
@@ -1,12 +1,115 @@
 package de.westnordost.streetcomplete.quests.surface
 
-sealed interface SidewalkSurfaceAnswer
+import de.westnordost.streetcomplete.data.osm.edits.update_tags.StringMapChangesBuilder
+import de.westnordost.streetcomplete.osm.Tags
+import de.westnordost.streetcomplete.osm.hasCheckDateForKey
+import de.westnordost.streetcomplete.osm.removeCheckDatesForKey
+import de.westnordost.streetcomplete.osm.surface.Surface
+import de.westnordost.streetcomplete.osm.surface.SurfaceAnswer
+import de.westnordost.streetcomplete.osm.updateCheckDateForKey
 
 data class SidewalkSurface(
     val left: SurfaceAnswer?,
     val right: SurfaceAnswer?,
-) : SidewalkSurfaceAnswer
+) : SidewalkSurfaceAnswer()
 
-object SidewalkIsDifferent : SidewalkSurfaceAnswer
+object SidewalkIsDifferent : SidewalkSurfaceAnswer()
 
 data class SidewalkSurfaceSide(val surface: Surface)
+
+open class SidewalkSurfaceAnswer {
+    fun applyTo(tags: StringMapChangesBuilder) {
```

Comment:
It's more consistent if the applyTo function was an extension function, separated from the enum / sealed interface.

Also, this is long enough / complex enough to go into an own file. I.e. `SidewalkSurfaceAnswerCreator.kt`

Reply:
hmm

in this case I just basically moved the entire file, but I can also refactor it

Reply:
Yes, please